Content.Server/Antag/Components/AntagSelectionComponent.cs

```diff
@@ -0,0 +1,96 @@
+using Content.Server.Destructible.Thresholds;
+using Content.Shared.Antag;
+using Content.Shared.Roles;
+using Content.Shared.Whitelist;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Antag.Components;
+
+[RegisterComponent]
+public sealed partial class AntagSelectionComponent : Component
+{
+    [DataField]
+    public bool SelectionsComplete;
+
+    [DataField]
+    public List<AntagSelectionDefinition> Definitions = new();
+
+    [DataField]
+    public List<(EntityUid, string)> SelectedMinds = new();
+
+    public AntagSelectionTime SelectionTime = AntagSelectionTime.PostPlayerSpawn;
+
+    public HashSet<ICommonSession> SelectedSessions = new();
+}
+
+[DataDefinition]
+public partial struct AntagSelectionDefinition
+{
+    [DataField]
+    public List<ProtoId<AntagPrototype>> PrefRoles = new();
+
+    [DataField]
+    public List<ProtoId<AntagPrototype>> FallbackRoles = new();
+
+    [DataField]
+    public AntagAcceptability MultiAntagSetting = AntagAcceptability.None;
+
+    [DataField]
+    public int Min = 1;
+
+    [DataField]
+    public int Max = 1;
+
+    [DataField]
+    public MinMax? MinRange;
+
+    [DataField]
+    public MinMax? MaxRange;
+
+    //todo implement a max range thing? make both of these ranges?
+
+    [DataField]
+    public int PlayerRatio = 10;
+
+    //todo: find out how to do this with minimal boilerplate: filler department, maybe?
+    //public HashSet<ProtoId<JobPrototype>> JobBlacklist = new()
+
+    /// <remarks>
+    /// Mostly just here for legacy compatibility and reducing boilerplate
+    /// </remarks>
+    [DataField]
+    public bool AllowNonHumans = false;
+
+    [DataField]
+    public EntityWhitelist? Whitelist;
+
+    [DataField]
+    public EntityWhitelist? Blacklist;
+
+    [DataField(serverOnly: true)]
+    public ComponentRegistry Components = new();
+
+    [DataField(serverOnly: true)]
```

Comment:
the file is in server code already so dont need (unless you want to make it shared down the line)